core/src/main/java/org/apache/iceberg/BaseScan.java

```diff
@@ -99,7 +99,7 @@ protected Schema tableSchema() {
     return schema;
   }
 
-  protected TableScanContext context() {
+  public TableScanContext context() {
```

Comment:
This should not be made public. I think we'll need to find a different way to report context.